### PR TITLE
fix: refactor useProvider hook

### PR DESCRIPTION
**Pull request by WhiteNik16**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- `@rarimo/react-provider`: refactor useProvider hook
```
Comment (napalmpapalam) [inline CHANGELOG.md:10]:
```md
- `@rarimo/react-provider`:  `useProvider` hook reactivity
```

File: packages/react-provider/src/hooks/useProvider.ts [hunk 2]
```diff
@@ -40,6 +40,24 @@ type ProviderEventPayload =
   | ProviderChainChangedEventPayload
   | ProviderInitiatedEventPayload
 
+type ProviderType = {
```
Comment (napalmpapalam) [inline packages/react-provider/src/hooks/useProvider.ts:43]: why did u create extra interface if u got almost the same in the provider package?

File: packages/react-provider/src/hooks/useProvider.ts [hunk 3]
```diff
@@ -56,30 +74,44 @@ type ProviderEventPayload =
...
+  const addChain = async (chain: Chain): Promise<void> =>
+    await provider?.addChain?.(chain)
+
+  const switchChain = async (chainId: ChainId): Promise<void> =>
+    await provider?.switchChain(chainId)
+
+  const signAndSendTx = async (
+    txRequestBody: TxRequestBody,
+  ): Promise<TransactionResponse | undefined> =>
+    await provider?.signAndSendTx?.(txRequestBody)
```
Comment (napalmpapalam) [inline packages/react-provider/src/hooks/useProvider.ts:99]: remove redundant `await`, u return a promise which client code will await anyway

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (WhiteNik16) [reply CHANGELOG.md:10]: fixed

File: packages/react-provider/src/hooks/useProvider.ts (hunk 2 above)
Comment (WhiteNik16) [reply packages/react-provider/src/hooks/useProvider.ts]: fixed

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: packages/react-provider/src/hooks/useProvider.ts [hunk 4]
```diff
@@ -53,33 +54,51 @@ type ProviderEventPayload =
  * @param providerProxy The type of provider from the @rarimo/provider package, such as {@link @rarimo/provider!MetamaskProvider}
  * @param createProviderOpts Options for the connection
  */
+
+interface UseProviderTypes extends ProviderBase{
+  provider: IProvider
+}
```
[the remark this answers is not in the document]
Comment (WhiteNik16) [reply packages/react-provider/src/hooks/useProvider.ts]: fixed

**Pull request merged**